src/generate-colors.ts

```diff
@@ -0,0 +1,25 @@
+import { generateTypeColors, generatePrimaryColors } from './utils/index'
+import { DEFAULT_COLOR_OPTIONS } from './types'
+import type { Colors, ColorOptions } from './types'
```

Comment:
I would suggest you to put actual value outside `types` file, since `.types` is meant for storing types.